packages/teleport/src/Apps/AddApp/useAddApp.ts

```diff
@@ -27,12 +28,19 @@ export default function useAddApp(ctx: TeleportContext) {
   const [automatic, setAutomatic] = useState(isEnterprise);
   const [cmd, setCmd] = useState('');
   const [expires, setExpires] = useState('');
+  const [token, setToken] = useState('');
+
+  useEffect(() => {
```

Comment:
automatic for apps works slightly different

how automatic was working before:
  - validate fields
  - fetch token

now it's doing:
  - fetch token
  - validate fields
  - fetch token

i think the proper way to handle this is on manual render, fetch token (so put this block in manual component)

Reply:
Thanks for pointing that out!